src/main/java/com/kaytat/simpleprotocolplayer/ThreadStoppable.java

```diff
@@ -0,0 +1,9 @@
+package com.kaytat.simpleprotocolplayer;
+
+class ThreadStoppable extends Thread {
+    boolean running = true;
+
```

Comment:
I really think you should add volatile here.  I see that it wasn't there originally but I think it should be added since customStop is called from the main service thread.

For reference:
http://stackoverflow.com/questions/106591/do-you-ever-use-the-volatile-keyword-in-java